taichi/backends/opengl/codegen_opengl.cpp

```diff
@@ -142,14 +144,14 @@ class KernelGen : public IRVisitor {
     if (used.int64)
       kernel_header += "layout(std430, binding = 0) buffer data_i64 { int64_t _data_i64_[]; };\n";
 
-    if (used.buf_gtmp) {
+    if (used.buf_gtmp || used.random) {
       kernel_header +=
-          "layout(std430, binding = 1) buffer gtmp_i32 { int _gtmp_i32_[]; };\n"
-          "layout(std430, binding = 1) buffer gtmp_f32 { float _gtmp_f32_[]; };\n";
+          "layout(std430, binding = 1) buffer gtmp_i32 { int _rand_state_[4]; int _gtmp_i32_[]; };\n"
```

Comment:
why `4`?

Reply:
Make `double` and `int64_t` alignment happy.